src/Writer/CsvWriter.php

```diff
@@ -58,21 +58,28 @@ class CsvWriter implements TypedWriterInterface
      */
     protected $withBom;
 
+    /**
+     * @var string
+     */
+    protected $terminate;
+
     /**
      * @param string $filename
      * @param string $delimiter
      * @param string $enclosure
      * @param string $escape
      * @param bool   $showHeaders
      * @param bool   $withBom
+     * @param string $terminate
      */
-    public function __construct($filename, $delimiter = ',', $enclosure = '"', $escape = '\\', $showHeaders = true, $withBom = false)
+    public function __construct(string $filename, string $delimiter = ',', string $enclosure = '"', string $escape = '\\', bool $showHeaders = true, bool $withBom = false, string $terminate = "\n")
     {
         $this->filename = $filename;
         $this->delimiter = $delimiter;
         $this->enclosure = $enclosure;
         $this->escape = $escape;
         $this->showHeaders = $showHeaders;
+        $this->terminate = $terminate;
```

Comment:
You are talking about how eacape parameter is not used but you are also adding terminate, you will have to do this in a separate PR. 1 PR = 1 fix/feature

Reply:
Will do.